yt_dlp/extractor/txxx.py

```diff
@@ -0,0 +1,307 @@
+import base64
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    int_or_none,
+    parse_duration,
+    qualities,
+)
+
+
+class TxxxBaseIE(InfoExtractor):
+    # some non-standard characters are used in the base64 string
+    _BASE64_CHAR_REPL_MAP = {
+        "\u0405": "S",
+        "\u0406": "I",
+        "\u0408": "J",
+        "\u0410": "A",
+        "\u0412": "B",
+        "\u0415": "E",
+        "\u041a": "K",
+        "\u041c": "M",
+        "\u041d": "H",
+        "\u041e": "O",
+        "\u0420": "P",
+        "\u0421": "C",
+        "\u0425": "X",
+        ",": "/",
+        ".": "+",
+        "~": "=",
+    }
+
+    def _decode_base64(self, text):
+        for from_char, to_char in self._BASE64_CHAR_REPL_MAP.items():
+            text = text.replace(from_char, to_char)
+        return base64.b64decode(text).decode('utf-8')
+
+    def _get_format_id(self, format_id):
+        if not format_id:
+            return None
+        elif isinstance(format_id, list):
+            return '' if len(format_id) == 0 else format_id[0].lstrip('_')
+        else:
+            return format_id.lstrip('_')
+
+
+class TxxxIE(TxxxBaseIE):
+    _VALID_URL = r'''(?x)
+                     https?://
+                     (?:(?:www\.)?(?P<host>
+                     hclips\.com|
+                     hdzog\.com|
+                     hotmovs\.com|
+                     inporn\.com|
+                     privatehomeclips\.com|
+                     tubepornclassic\.com|
+                     txxx\.com|
+                     txxx\.tube|
+                     upornia\.com|
+                     vjav\.com|
+                     vxxx\.com|
+                     voyeurhit\.com|
+                     voyeurhit\.tube))
+                     (?:/(?:video/|videos/|video-|embed/)(?P<id>[^/]+)/)
+                  '''
```

Comment:
indent by nesting

Reply:
Do you mean to intend like this instead?

```
    _VALID_URL = r'''(?x)
        https?://
        (?:(?:www\.)?(?P<host>
```